### ISSUE TITLE
Assigning null to missing ValueTuple type yields assertions

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#13088]
```C#
    [Fact]
        public void AssignNullWithMissingValueTuple()
        {
            var source = @"
public struct S
{
    (int, int) t = null;
}
";

            var comp = CreateCompilationWithMscorlib(source);
            comp.VerifyDiagnostics(); // crashes
        }
```

FYI @VSadov

**Comment by VSadov** [comment]
Either assert is bad or we have corrupted something. 
Need to figure what is going on.

**Comment by jcouv** [comment]
Fixed by #13258
